### ISSUE TITLE
[FN] UnusedLocalVariable после вставки в структуру

### ISSUE THREAD

**Issue by asosnoviy** [1c-syntax/bsl-language-server#2787]
## Диагностика
UnusedLocalVariable

## Описание ложного НЕ срабатывания диагностики

Создана структура, в нее добавлены ключи, но структура никуда не передана

## Пример кода

```bsl

	ПараметрыПроцедуры = Новый Структура;
	ПараметрыПроцедуры.Вставить("Проблема", Проблема);
	ПараметрыПроцедуры.Вставить("ПараметрыПроверки", ПараметрыПроверки);

```

**Comment by asosnoviy** [comment]
Да здравствует выведение типов!
https://github.com/1c-syntax/bsl-language-server/issues/269

**Comment by artbear** [comment]
Сводная задача по "потерянным" переменным https://github.com/1c-syntax/bsl-language-server/issues/1088

**Comment by asosnoviy** [comment]
Мысль где то в недрах референс индекса помечать "самоиспользование".

**Comment by artbear** [comment]
Да, проверка самоиспользования внутри реф-индекса будет очень полезной.

без нее мне уже в 2х или 3х правилах пришлось костылить и самому писать код проверки самозаписи
- Значение = Значение

**Comment by artbear** [comment]
@qtLex что скажешь?

> Да, проверка самоиспользования внутри реф-индекса будет очень полезной.
> 
> без нее мне уже в 2х или 3х правилах пришлось костылить и самому писать код проверки самозаписи
> 
> * Значение = Значение

**Comment by artbear** [comment]
Я на днях реализовал 2 правила
- https://github.com/1c-syntax/bsl-language-server/pull/2810 (готово)
- https://github.com/1c-syntax/bsl-language-server/issues/1088 (заканчиваю)

в обоих случаях мне понадобилось различать 
- самоприсвание `Значение = Значение;`
- самоприсваивание с переиспользованием - `Значение = ЛюбоеВыражениеВКоторомЕстьЗначение;`
- использование идет уже после присваивания `Значение = 10; ЛюбоеВыражениеВКоторомЕстьЗначение...;`

минусы
- пока приходится дублировать код
- главный минус в том, что нужно повторно бегать по дереву и определять каждый из этих признаков

по опыту написания различных хитрых правил я знаю, что подобные проверки часто приходится повторять
- например, в правиле для ЖР, СтрШаблон и т.п.

т.е. было бы хорошо, чтобы была возможность определения указанных признаков связи определения и использовния
- и кажется, что эти признаки не зависят от вида переменной
